src/components/Balance.tsx

```diff
@@ -16,7 +16,15 @@ interface BalanceProps extends TextProps {
 }
 
 const StyledText = styled(Text)<TextProps>`
-  color: ${({ isDisabled, color, theme }) => (isDisabled ? theme.colors.textDisabled : color)};
+  color: ${({ isDisabled, color, theme }) => {
+    if (isDisabled) {
+      return theme.colors.textDisabled
+    }
+    if (!color) {
```

Comment:
🤔 
```
   if (!color) {
      return color
    }
```

Reply:
this change doesn't seem useful because the color already default to `text`

Reply:
You are right, don't remember why i put it there :D. I changed now, i am checking against default color value if there is no value coming from component props (which in this case) I fallback to theme text color .

Reply:
For me, this change does exactly the same thing than the code before, but with more code. I'm not sure I understand what you're trying to do

Reply:
Issue is when you switch to dark theme, it uses the same color value which is ‘text’ instead of theme one  unless it is passed explicitly. Thats why I added a theme.color.text. It is reproducible in iOS.

Reply:
`text` and `theme.color.text` are the same

Reply:
Good news, I think i found the issue, when we use styledtext with css attributes, it overrides color attribute of actual dom object and interpreted as color: 'text' or color: 'textdisabled' instead of actual html color values. Browsers failed and fallback to other color and in some browsers (metamask, safari) it is different than we expect. Thats why I removed styledText and put condition in Text props.